OloEngine/src/OloEngine/Particle/ParticleSystem.cpp

```
@@ -0,0 +1,518 @@
+                m_Pool.m_Lifetimes[idx] = lifetime;
+                m_Pool.m_MaxLifetimes[idx] = lifetime;
+
+                if (TrailModule.Enabled)
+                {
+                    m_TrailData.ClearTrail(idx);
+                }
+            }
+        }
+    }
+
+    void ParticleSystem::SortByDepth(const glm::vec3& cameraPosition)
```

Clang (14.0.6)

[warning] method 'SortByDepth' can be made static

(readability-convert-member-functions-to-static)